### DISCUSSION TITLE
Versioning - the correct way

### DISCUSSION THREAD

**Discussion by ondrej-krc-jediny** [CommunityToolkit/Maui#2416] [Q&A] [answered]
I have the newest Visual Studio 17.12.3 installed, which comes with MAUI workload: 
`dotnet workload list`
`Installed Workload Id      Manifest Version       Installation Source`
`android                    35.0.24/9.0.100        SDK 9.0.100, VS 17.12.35527.113`
`ios                        18.2.9170/9.0.100      SDK 9.0.100, VS 17.12.35527.113`
`maccatalyst                18.2.9170/9.0.100      SDK 9.0.100, VS 17.12.35527.113`
`maui-windows               9.0.0/9.0.100          SDK 9.0.100, VS 17.12.35527.113`

I retargeted my MAUI app to .NET 9, now I'd like to use CommunityToolkit 10.0.0. When I try updating the NuGet package, I get these errors:

`Warning As Error: Detected package downgrade: Microsoft.Maui.Controls from 9.0.21 to 9.0.0. Reference the package
      directly from the project to select a different version.
       MobileApp -> CommunityToolkit.Maui 10.0.0 -> Microsoft.Maui.Controls (>= 9.0.21 && < 10.0.0)
       MobileApp -> Microsoft.Maui.Controls (>= 9.0.0)`

I reference MAUI Controls this way in my .csproj:

`<PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />`

If I hardcode the version:

`<PackageReference Include="Microsoft.Maui.Controls" Version="9.0.21" />`

It works, but I don't like this approach, because I need to think about updating it every time I update Visual Studio plus there is an inconsistency between MAUI workload version and version and the MAUI NuGet package version.

What is the clean way to approach this please?
Dotnet workload update does nothing - preserves the version listed above.

**Comment by VladislavAntonyuk** [comment, marked as answer]
we require 9.0.21 because it contains important fixes for us. Once VS updates MAUI workload up to 9,0.21 you can use $(MauiVersion)
